lib/AST/ASTContext.cpp

```diff
@@ -4250,6 +4250,17 @@ FunctionType *FunctionType::get(ArrayRef<AnyFunctionType::Param> params,
   return funcTy;
 }
 
+#ifndef NDEBUG
+static bool isConsistentAboutIsolation(const llvm::Optional<ASTExtInfo> &info,
+                                       ArrayRef<AnyFunctionType::Param> params) {
+  bool hasIsolatedParameter = false;
+  for (auto param : params) {
+    if (param.isIsolated()) hasIsolatedParameter = true;
+  }
+  return (hasIsolatedParameter == (info && info->getIsolation().isParameter()));
```

Comment:
Could be simplified down to:

```return (info && info->getIsolation().isParameter()) == hasIsolatedParameter(params);```

Reply:
Ah right, I added that function after I hand-rolled some of these loops.